cura/Scene/ConvexHullDecorator.py

```diff
@@ -229,15 +230,53 @@ def _compute2DConvexHull(self):
             return offset_hull
 
     def _getHeadAndFans(self):
-        return Polygon(numpy.array(self._global_stack.getProperty("machine_head_with_fans_polygon", "value"), numpy.float32))
+        original_head_and_fans = self._global_stack.getProperty("machine_head_with_fans_polygon", "value")
+        full_area = copy.deepcopy(original_head_and_fans)
+
+        # In one at a time mode, when multiple extruders or not the 1st extruder is used, the head and fan convex hull
+        # needs to take into account the extruder offset(s) as well so it will not collide into a printed model.
+        # In this case, we calculate the head and fan convex hull by taking the union of all those used extruders.
+        # If, for example, only the 2nd extruder is used, we will still take into account the 1st extruder.
+        machine_manager = Application.getInstance().getMachineManager()
+        extruder_manager = ExtruderManager.getInstance()
+        # Take into account the extruder offset for creating the head-and-fans polygon
+        used_extruders = extruder_manager.getUsedExtruderStacks()
+        max_used_extruder_position = max(int(e.getMetaDataEntry("position")) for e in used_extruders)
+        multiple_extruders_in_use = max_used_extruder_position > 0
+        for extruder_position in range(max_used_extruder_position + 1):
+            extruder = machine_manager.activeMachine.extruders[str(extruder_position)]
+            offset_x = extruder.getProperty("machine_nozzle_offset_x", "value")
+            offset_y = extruder.getProperty("machine_nozzle_offset_y", "value")
+
+            offset_polygon = copy.deepcopy(original_head_and_fans)
+
+            def apply_offset(rect, offset):
+                for point in rect:
+                    point[0] += offset[0]
+                    point[1] += offset[1]
+
+            apply_offset(offset_polygon, [-offset_x, -offset_y])
+
+            def union(rect1, rect2):
+                bl = [min(rect1[0][0], rect2[0][0]), min(rect1[0][1], rect2[0][1])]
+                tl = [min(rect1[1][0], rect2[1][0]), max(rect1[1][1], rect2[1][1])]
+                tr = [max(rect1[2][0], rect2[2][0]), max(rect1[2][1], rect2[2][1])]
+                br = [max(rect1[3][0], rect2[3][0]), min(rect1[3][1], rect2[3][1])]
+                return [tl, bl, tr, br]
```

Comment:
What is it - **tl, bl, tr, br** ?

Reply:
top-left
bottom-left
top-right
bottom-right

Reply:
There were some mistakes. Fixed in 42b86086c